app/views/unattended/provisioning_templates/cloud_init/cloud_init_default.erb

```diff
@@ -24,10 +24,12 @@ hostname: <%= @host.name %>
 fqdn: <%= @host %>
 manage_etc_hosts: true
 users: {}
-<% if puppet_enabled %>
 runcmd:
 - |
-<% if host_param_true?('enable-puppetlabs-pc1-repo') || host_param_true?('enable-puppetlabs-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
+<%= indent(2) { snippet 'redhat_register' } %>
+- |
+<% if puppet_enabled %>
+<% if host_param_true?('enable-puppetlabs-repo') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
 <%= snippet 'puppetlabs_repo' %>
 <% end -%>
 <%= indent(2) { snippet('puppet_setup', :variables => { :full_puppet_run => true }) } %>
```

Comment:
I believe several lines should have `-%>` instead to prevent those empty lines which might (or might not) break the YAML. It's probably better to have that formatted nicely, I am going to file a fix into community-templates repo.

Reply:
Fixed here: https://github.com/theforeman/community-templates/pull/646